### PR TITLE
oep(volume): propose enhancements for all cstorvolumes related APIs

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

### CODE REVIEW

**Review by vishnuitta — APPROVED** [review]
looks good..
just one comment as mentioned in OEP PR: DRF of CStorVolume.spec as RF, and, RF of CStorVolume.spec into its Status.

File: design/cstor/v1/cstor_volume.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
+# Move `CStorVolumes`, `CStorVolumeReplica` resources Schema to version `v1`
```
Comment (kmova) [inline design/cstor/v1/cstor_volume.md]: @prateekpandey14, can you follow the same format as the cstorpool.md. 

This document should just contain the new v1 API schema. No need to mention the old schema or changes from the earlier schema is this PR. 

Same comments apply to the other file as well.

Comment (kmova) [reply design/cstor/v1/cstor_volume.md]: Add the schema for cStorVolumeAttachment as well.

Comment (prateekpandey14) [reply design/cstor/v1/cstor_volume.md]: done

**Comment by prateekpandey14** [comment]
I think we should be having a directory structure like `docs/design/cstor/....`. `docs` will be the main dir and then arrange all the files under that.

File: design/cstor/v1/cstorvolume.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
+# Proposed v1 CStorVolumes and CStorVolumeReplica Schema
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Schema Proposal](#volume-schema-proposal)
```
Comment (kmova) [inline design/cstor/v1/cstorvolume.md]: n_: change the title to same title as below

File: design/cstor/v1/cstorvolume.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+## Introduction
+
+This proposal highlights the enhancements and migration of current `CStorVolumes` and `CStorVolumeReplicas` schema and
+proposes improvements.
+
+# Goals
+
+The major goal of this document is to freeze the APIs schema for `CStorVolumes` and `CStorVolumeReplica`.
+Apart from this the document focuses on following aspects too:
```
Comment (kmova) [inline design/cstor/v1/cstorvolume.md]: Remove these links. the only goal is to propose v1 schema for `CStorVolumes` and `CStorVolumeReplica`

File: design/cstor/v1/cstorvolume.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+- Migrating the `CStorVolumes` and `CStorVolumeReplica` CRD to `v1` apiversion.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumes` and `CStorVolumeReplica`.
+- Enhancements for operations based resource events based in `CStorVolumes` and `CStorVolumeReplicas`
+
+# Enhancements
```
Comment (kmova) [inline design/cstor/v1/cstorvolume.md]: Remove this section

File: design/cstor/v1/cstorvolume_policy.md [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [VolumeConfig Schema Proposal](#volume-config-schema-proposal)
+* [VolumePolicy Schema Proposal](#volume-policy-schema-proposal)
+* [VolumeAttachment Schema Proposal](#volume-attachment-schema-proposal)
+
+## Introduction
+
+This proposal highlights enhancements and migration of current `CStorVolumeConfig` and `CStorVolumePolicy` schema and
```
Comment (kmova) [inline design/cstor/v1/cstorvolume_policy.md:13]: Same as the previous doc. The only goal is to propose the schema of various CRs.

File: design/cstor/v1/cstorvolume.md (hunk 4 above)
Comment (prateekpandey14) [reply design/cstor/v1/cstorvolume.md]: done

File: design/cstor/v1/cstorvolume.md (hunk 3 above)
Comment (prateekpandey14) [reply design/cstor/v1/cstorvolume.md]: done

File: design/cstor/v1/cstorvolume.md (hunk 2 above)
Comment (prateekpandey14) [reply design/cstor/v1/cstorvolume.md]: done

**Review by kmova — APPROVED** [review]

**Pull request merged**